custom_model_runner/datarobot_drum/drum/memory_monitor.py

```diff
@@ -22,10 +22,33 @@ def __init__(self, pid=None, include_childs=True):
         self._include_childs = include_childs
 
     def collect_memory_info(self):
-        current_process = psutil.Process(os.getpid())
-        proc_rss_mb = ByteConv.from_bytes(current_process.memory_info().rss).mbytes
-        for child in current_process.children(recursive=True):
-            proc_rss_mb += ByteConv.from_bytes(child.memory_info().rss).mbytes
+        def get_proc_data(p):
+            return {
+                "pid": p.pid,
+                "cmdline": p.cmdline(),
+                "mem": ByteConv.from_bytes(p.memory_info().rss).mbytes,
+            }
+
+        drum_process = None
+        nginx_rss_mb = 0
+        drum_rss_mb = 0
+
+        for proc in psutil.process_iter():
+            if "drum" in proc.name().lower():
+                if "server" in proc.cmdline():
+                    drum_process = proc
```

Comment:
isn't this code running in the drum process? why not get your own pid?

Reply:
Also it does seem brittle here

Reply:
When running with uwsgi this code runs in a worker, which is drum's second or third child.
So I search for drum process and take it's children.

nginx is not drum's child, so scan it separately.

Reply:
You can get parent pid more than once.

Reply:
**DONE**